langchain/evaluation/qa/chat_eval_chain.py

```diff
@@ -0,0 +1,95 @@
+"""LLM Chain specifically for evaluating question answering."""
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import Any, Dict, List
+
+from pydantic import BaseModel
+
+from langchain.chat.base import BaseChatChain
+from langchain.chat_models.base import BaseChatModel
+from langchain.evaluation.qa.eval_prompt import (
+    CHAT_INSTRUCTIONS,
+    CHAT_RESPONSE_TEMPLATE,
+)
+from langchain.schema import ChatMessage
+
+
+class ChatPromptTemplate(BaseModel, ABC):
+    input_variables: List[str]
+
+    @abstractmethod
+    def format(self, *, query: str, answer: str, result: str) -> List[ChatMessage]:
+        """Format chat prompt template."""
```

Comment:
Shouldn't this class be somewhere else and be more generic than this?

Reply:
yup